### ISSUE TITLE
Unit test AsciidoctorExtension

### ISSUE THREAD

**Issue by rmpestano** [cukespace/arquillian-asciidoctor-extension#4]
As commented [here](https://github.com/cukespace/arquillian-asciidoctor-extension/issues/1#issuecomment-243885815) it would be nice to have an easy way to test the extension (current approach involves[ ordering two tests](https://github.com/cukespace/arquillian-asciidoctor-extension/blob/master/pom.xml#L179), one for adocGeneration and another for assertions).

I've taked to arquillan team guys and they've suggested something[ like this](https://github.com/rmpestano/arquillian-asciidoctor-extension/blob/e53bb37f9b7fd1659d7466df1a3e9725bd85797b/src/test/java/com/github/cukespace/arquillian/asciidoctor/AdocExtensionTest.java#L38) but unfortunately its currently not working (events are not being fired in the observer), I have used [this cube test](https://github.com/arquillian/arquillian-cube/blob/master/core/src/test/java/org/arquillian/cube/impl/client/container/CubeContainerLifecycleControllerTest.java) as reference.

@bartoszmajsak is going to look when he get some time, meanwhile the test is marked as ignored.

What do you think?

**Comment by rmannibucau** [comment]
I don't want of any mock there cause I saw such tests breaking often enough to have 0 trust in them. The cost of the 2 executions is very small IMO and easy enough to use in an IDE (run arquillian one then the standard one). Alternative is to write a custom runner start/stopping arquillian then running asserts. It can actually be better but having 2 executions allows us to test the state after surefire execution which is good and avoid to have leaked anything in the surefire fork.

In short: think we are good enough for now but happy to also get a AroundArquillianRunner if you think it is easier to write.

**Comment by rmpestano** [comment]
Hi, will not enter on an endless discussion about 'mock or not to mock' so lets stay with the more practical solution, the current one. Thanks for the comment, will close the issue.
